lib/dor/services/registration_service.rb

```diff
@@ -18,23 +59,18 @@ def register_object(params = {})
         raise Dor::ParameterError, "Unknown item type: '#{object_type}'" if item_class.nil?
 
         # content_model = params[:content_model]
-        admin_policy  = params[:admin_policy]
+        # parent        = params[:parent]
         label         = params[:label]
         source_id     = params[:source_id] || {}
         other_ids     = params[:other_ids] || {}
         tags          = params[:tags] || []
-        # parent        = params[:parent]
         collection    = params[:collection]
-        pid = nil
-        if params[:pid]
-          pid = params[:pid]
-          existing_pid = SearchService.query_by_id(pid).first
-          unless existing_pid.nil?
-            raise Dor::DuplicateIdError.new(existing_pid), "An object with the PID #{pid} has already been registered."
-          end
-        else
-          pid = Dor::SuriService.mint_id
-        end
+
+        # Check for sourceId conflict *before* potentially minting PID
+        source_id_string = check_source_id [source_id.keys.first, source_id[source_id.keys.first]].compact.join(':')
```

Comment:
this resolution from source_id argument into a useful "abc:xyz" string is hard to read and uses source_id.keys.first twice -- maybe you should also do the fail `ArgumentError` before this line.